utils/netutils/netutils.go

```diff
@@ -48,8 +50,105 @@ func init() {
 	}
 }
 
+func ipv6ToInt(IPv6Addr string) *big.Int {
+	IPv6Address := net.ParseIP(IPv6Addr)
+	IPv6Int := big.NewInt(0)
+	IPv6Int.SetBytes(IPv6Address.To16())
+	return IPv6Int
+}
+
+type ipAddress []byte
+
+const hexDigit = "0123456789abcdef"
+
+func getIPv6Range(Subnetv6 string) (string, string) {
+	_, ipv6IP, _ := net.ParseCIDR(Subnetv6)
+
+	subnetMask := ipv6IP.Mask
+	subnetStartRange := ipv6IP.IP
+
+	for i := 0; i < 16; i++ {
+		subnetMask[i] = ^subnetMask[i]
+	}
+	n := len(subnetStartRange)
+
+	subnetEndRange := make(ipAddress, n)
+	for i := 0; i < n; i++ {
+		subnetEndRange[i] = subnetStartRange[i] | subnetMask[i]
+	}
+
+	return subnetStartRange.String(), subnetEndRange.byteToString()
+}
+
+// IsOverlappingSubnetv6 verifies the Overlapping of subnet for v6 networks
+func IsOverlappingSubnetv6(inputIPv6Subnet string, existingIPv6Subnet string) bool {
+	inputIPv6StartRange, inputIPv6EndRange := getIPv6Range(inputIPv6Subnet)
+	existingIPv6StartRange, existingIPv6EndRange := getIPv6Range(existingIPv6Subnet)
+
+	inputStartRange := ipv6ToInt(inputIPv6StartRange)
+	inputEndRange := ipv6ToInt(inputIPv6EndRange)
+
+	existingStartRange := ipv6ToInt(existingIPv6StartRange)
+	existingEndRange := ipv6ToInt(existingIPv6EndRange)
+
+	if (existingStartRange.Cmp(inputStartRange) == -1 && inputStartRange.Cmp(existingEndRange) == -1) ||
+		(existingStartRange.Cmp(inputEndRange) == -1 && inputEndRange.Cmp(existingEndRange) == -1) ||
+		(existingStartRange.Cmp(inputStartRange) == 0) ||
+		(inputStartRange.Cmp(existingEndRange) == 0) ||
+		(existingStartRange.Cmp(inputEndRange) == 0) ||
+		(inputEndRange.Cmp(existingEndRange) == 0) {
+		return true
+	}
+
+	if (inputStartRange.Cmp(existingStartRange) == -1 && existingStartRange.Cmp(inputEndRange) == -1) ||
+		(inputStartRange.Cmp(existingEndRange) == -1 && existingEndRange.Cmp(inputEndRange) == -1) {
+		return true
+	}
+	return false
+}
+
+// String returns the hexadecimal form of m, with no punctuation.
+func (m ipAddress) byteToString() string {
```

Comment:
Please see if you can reuse an existing function from golfing `net` pkg.
I'm not convinced you need a new type for `ipAddress`. you should be able to reuse `net.IP` type